File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/EventSupply.swift
import Foundation

// MARK: - SupplySource

public enum SupplySource: String, Codable, Sendable, CaseIterable, Hashable {
    case stock
    case purchase
}

// MARK: - EventSupply

public struct EventSupply: Codable, Identifiable, Sendable, Hashable {
    public let id: String
    public let eventId: String
    public let inventoryId: String
    public let quantity: Double
    public let unitCost: Double
    public let source: SupplySource
    public let excludeCost: Bool
    public let createdAt: String
    // Denormalized joined fields
    public var supplyName: String?
    public var unit: String?
    public var currentStock: Double?
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/EventSupply.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EventSupply(
    val id: String = "",
    @SerialName("event_id") val eventId: String = "",
    @SerialName("inventory_id") val inventoryId: String,
    val quantity: Double,
    @SerialName("unit_cost") val unitCost: Double,
    val source: SupplySource = SupplySource.STOCK,
    @SerialName("exclude_cost") val excludeCost: Boolean = false,
    @SerialName("created_at") val createdAt: String = "",
    @SerialName("supply_name") val supplyName: String? = null,
    val unit: String? = null,
    @SerialName("current_stock") val currentStock: Double? = null
)

@Serializable
enum class SupplySource {
    @SerialName("stock") STOCK,
    @SerialName("purchase") PURCHASE
}
